### PR TITLE
Split package on depencies from AUR

### PR DESCRIPTION
**Pull request by aboe76**
We shipped in the package ruby-hiera-json, ruby-hiera and vim-puppet
removed ruby-hiera, ruby-hiera-json and ruby-hiera-puppet

Ruby-hiera and ruby-hiera-json are already packaged in aur, just depend on them.
ruby-hiera-puppet is a conflict because puppet3 ships ruby-hiera-puppet.

conflicts with vim-puppet because we ship vim files with the package.

This makes a smaller build and easier to maintain.

### CODE REVIEW

File: puppet/PKGBUILD [hunk 1]
```diff
@@ -8,44 +8,34 @@
...
 
 # RC style, reserved for later use
-#pkgname=puppet
+#pkgname=puppet3
 #_rc=8
 #_pkgver=3.0.0
 #pkgver=${_pkgver}_rc${_rc}
 
-pkgname=puppet
+pkgname=puppet3
```
Comment (GregSutcliffe) [inline puppet/PKGBUILD]: Shouldn't this stay as 'puppet' ?

File: puppet/PKGBUILD [hunk 2]
```diff
@@ -8,44 +8,34 @@
...
 license=("APACHE")
-depends=("ruby" "ruby-shadow" "facter")
-optdepends=("ruby-augeas")
+depends=("ruby" "ruby-shadow" "facter" "ruby-hiera" "ruby-hiera-json")
 backup=("etc/puppet/puppet.conf")
 options=(emptydirs)
-provides=("puppet")
-conflicts=("puppet3")
+provides=("puppet" "vim-puppet" "ruby-hiera-puppet")
+conflicts=("puppet" "vim-puppet" "ruby-hiera-puppet")
```
Comment (GregSutcliffe) [inline puppet/PKGBUILD]: no point conflicting with itself - this should be puppet3

**Comment by GregSutcliffe** [comment]
Looks fine to me apart from minor puppet/puppet3 cosmetic changes :+1:

**Comment by aboe76** [comment]
That's true, puppet shouldn't conflict with itself, but it should conflict
with ruby-puppet, it is also in aur. and it installs puppet from a gem.

2012/11/19 Greg Sutcliffe notifications@github.com

**Comment by GregSutcliffe** [comment]
This got merged in d3057f7fefd6c54ebbb0ed2d059c37ef19c7f48f, forgot to close it :)

**Pull request closed without merging**